tt_metal/tt_stl/aligned_allocator.hpp

```diff
@@ -43,7 +43,7 @@ class aligned_allocator {
         return static_cast<T*>(p);
     }
 
-    void deallocate(T* p, std::size_t size) const noexcept { ::operator delete[](p, size * sizeof(T), alignment); }
+    void deallocate(T* p, std::size_t size) const noexcept { ::operator delete[](p, alignment); }
```

Comment:
why is this change needed?

Reply:
It wasn't present in libstdc++'s header, but after a bit more digging it seems like it can be enabled if we compile with flag ` -fsized-deallocation`.  In reading cppref I can't tell that there's any difference between APIs 3-4 and 7-8. https://en.cppreference.com/w/cpp/memory/new/operator_delete

Reply:
I can update the PR to use `-fsized-deallocation` if that seems better.